runtime/integration-tests/src/generic/envs/runtime_env.rs

```diff
@@ -92,7 +96,18 @@ impl<T: Runtime> Env<T> for RuntimeEnv<T> {
 			utils::create_extrinsic::<T>(who, call, nonce)
 		});
 
-		self.parachain_state_mut(|| T::Api::apply_extrinsic(extrinsic).unwrap())?;
+		self.parachain_state_mut(|| {
+			let res = T::Api::apply_extrinsic(extrinsic);
+			// NOTE: This is our custom error that we are having in the
+			//       `PreBalanceTransferExtension` SignedExtension, so we need to
+			//        catch that here.
```

Comment:
Why are we having this error? Should we configure something to avoid generating this error?

Reply:
The `PreBalancesTransferExtension` is filtering out all calls that are `pallet-balances::Transfer*` alike calls, if there is a restriction set and the receiver is not correct. So i.e. the filter for native currency. 

But it is not working on the DispatchError level, but on the validity level, so we can not return the correct errors.

Reply:
Ok, thanks for the explanation

Reply:
So, even if such an error is dispatched, the extrinsic is considered submitted, right?

Reply:
yes.